Fix: GH Action schedule failed to create promotion ticket (#8041) (#8110)
DataBiosphere/azul: UPGRADING.rst

reverted. This is all fairly informal and loosely defined. Hopefully we won't
have too many entries in this file.

#8041 Fix: GH Action schedule failed to create promotion ticket
===============================================================

Operator
--------

The following steps should be followed for the ``dev`` GitLab instance. Steps 1
and 2 also apply to ``anvildev``, but with a different interval pattern.

Edit the existing schedule in the Azul project on that ``dev`` GitLab instance.
Its description is ``Create Azul scheduled issues on GitHub``. You may need to
ask a system administrator to perform these changes on your behalf.

1) Set the cron timezone to ``Pacific Time (US & Canada)``

2) Set the variable ``azul_gitlab_schedule`` to ``github_schedule``

3) Set the interval pattern to ``0 * * * *``

Repeat steps 1 and 2 from above for ``anvildev``, then, set the interval pattern
to ``30 * * * *``.

Remove the ``AZUL_GITHUB_TOKEN`` from the GitHub Action secrets. You may need to

> ask a system administrator to perform this removal on your behalf.

Reviewer: Please request PL if you think I've misunderstood something.

```suggestion
You may need to ask a system administrator to perform these changes on your behalf.

Edit the ``github_schedule`` pipeline schedule on the ``dev`` and ``anvildev`` GitLab instances:

1) Set the description to "Create Azul scheduled issues on GitHub". 

2) Set the interval pattern to ``0 * * * *`` on ``dev`` and ``30 * * * *`` on ``anvildev``.

3) Set the cron timezone to ``Pacific Time (US & Canada)``

4) Set the variable ``azul_gitlab_schedule`` to ``github_schedule``

Next, remove ``AZUL_GITHUB_TOKEN`` from the GitHub Action secrets.
```

reverted. This is all fairly informal and loosely defined. Hopefully we won't
have too many entries in this file.

#8041 Fix: GH Action schedule failed to create promotion ticket
===============================================================

Operator
--------

You may need to ask a system administrator to perform these changes on your
behalf.

Add a new ``github_schedule`` pipeline schedule on the ``dev`` and ``anvildev``

> GitLab instances:

Reviewer: The new pipeline is added when the GitLab component is deployed, correct? So I think [my original wording](https://github.com/DataBiosphere/azul/pull/8110#discussion_r3489764900) is more accurate.

Author: >The new pipeline is added when the GitLab component is deployed, correct?

I don't think so, the GitLab schedule pipeline does not exist, as a schedule to be ran in the instance, until is manually created by a user. And this PR does not require deploying the GitLab component. However, I understand how this misconception may arise due to the added directive (`github_schedule`) in `.gitlab-ci.yml`, but this are just the instructions for a given job (or schedule).

Reviewer: You're right, I misunderstood the changes to `.gitlab-ci.yml`

1) Set the description to "Create Azul scheduled issues on GitHub".

2) Set the interval pattern to ``0 * * * *`` on ``dev`` and ``30 * * * *`` on ``anvildev``.

3) Set the cron timezone to ``Pacific Time (US & Canada)``

4) Set the variable ``azul_gitlab_schedule`` to ``github_schedule``

Next, confirm with SA that the ``azul_github_access_token`` in
``environment.local.py``, for both GL deployments is set to a token with similar

> permissions to the ``AZUL_GITHUB_TOKEN`` in the GitHub Action secrets.

Reviewer: What does "similar" mean here?
```suggestion
Next, confirm with the SA that the ``azul_github_access_token`` in
``environment.local.py`` is set to a token with similar
permissions to the ``AZUL_GITHUB_TOKEN`` in the GitHub Action secrets
for both GitLab deployments.
```

Author: Replaced it with "equivalent", to indicate that the same level of permissions are required.

reverted. This is all fairly informal and loosely defined. Hopefully we won't
have too many entries in this file.

#8041 Fix: GH Action schedule failed to create promotion ticket
===============================================================

Operator
--------

You may need to ask a system administrator to perform these changes on your
behalf.

Add a new ``github_schedule`` pipeline schedule on the ``dev`` and ``anvildev``
GitLab instances:

1) Set the description to "Create Azul scheduled issues on GitHub".

2) Set the interval pattern to ``0 * * * *`` on ``dev`` and ``30 * * * *`` on
``anvildev``.

> 3) Set the cron timezone to ``Pacific Time (US & Canada)``

Reviewer 2: ```suggestion
3) Set the Cron timezone to ``Pacific Time (US & Canada)``
```

4) Set the variable ``azul_gitlab_schedule`` to ``github_schedule``

Next, confirm with the SA that the ``azul_github_access_token`` used in both
instances' ``environment.local.py`` is set, and that its permissions are

> equivalent to the ``AZUL_GITHUB_TOKEN`` in the GitHub Action secrets.

Reviewer 2: ```suggestion
equivalent to that of the token deposited in the ``AZUL_GITHUB_TOKEN`` GitHub Action secret.
```

> Lastly, remove ``AZUL_GITHUB_TOKEN`` from the GitHub Action secrets.

Reviewer 2: I'm not sure that operators have the necessary permissions to do this.

reverted. This is all fairly informal and loosely defined. Hopefully we won't
have too many entries in this file.

#8041 Fix: GH Action schedule failed to create promotion ticket
===============================================================

Operator
--------

You may need to ask a system administrator to perform these changes on your
behalf.

> Add a new ``github_schedule`` pipeline schedule on the ``dev`` and ``anvildev``

Reviewer 2: It would have been helpful to describe where in the UI this is.

> still run the job. The following steps should be followed for the ``dev``

Reviewer: The first sentence feels out of place here. I would remove it. If you think it's important, it needs to be made clearer why someone following these instructions would need to know this.

GitLab instances:

1) Set the description to "Create Azul scheduled issues on GitHub".

> 2) Set the interval pattern to ``0 * * * *`` on ``dev`` and ``30 * * * *`` on

Reviewer 2: `2)` and `3)` should have been swapped since that's the order in which they occur in the UI

``anvildev``.

3) Set the Cron timezone to ``Pacific Time (US & Canada)``

4) Set the variable ``azul_gitlab_schedule`` to ``github_schedule``

> Next, confirm with the SA that the ``azul_github_access_token`` used in both

Reviewer 2: This should occur first in the instructions, as it needs to be done *before* the schedule is created.

Repeat steps 1 and 2 from above for ``anvildev``, then, set the interval pattern
to ``30 * * * *``.

Retiring the GitHub Actions workflow means dropping the GitHub schedule. Remove
the ``AZUL_GITHUB_TOKEN`` from the GitHub Action secrets as a cleanup step. You

> may need to ask a system administrator to perform this removal on your behalf.

Reviewer: Again, oddly verbose. 
```suggestion
Remove the ``AZUL_GITHUB_TOKEN`` from the GitHub Action secrets. You
may need to ask a system administrator to perform this removal on your behalf.
```
